src/openlrr/game/front/Loader.cpp

```diff
@@ -22,18 +26,241 @@ LegoRR::Loader_Globs & LegoRR::loaderGlobs = *(LegoRR::Loader_Globs*)0x00556e00;
+	}
+
+	//Check if profile is empty and open file
+	if (profileName != nullptr) {
+		f = Gods98::File_Open(profileName, "r");
+	}
+
+	//Iterate through all loader sections and store to loaderStruct
+	localLoaderSec = loaderGlobs.sectionList;
+	do {
+		//Check if file/string is null or if we have a string that isnt XXX=YYY
+		if ( (f == nullptr) || (Gods98::File_GetS(buff, BUFFSIZE, f) == nullptr) || (Gods98::Util_Tokenise(buff, stringParts, "=") != 2) )
+		{
+			localLoaderSec->name = nullptr;
+		}
+		else {
+
+			//Allocate memory for the string on the struct and copy data over
+			uint32* allocPtr = (uint32*)Gods98::Mem_Alloc(strlen(stringParts[0]));
+			localLoaderSec->name = (char*)allocPtr;
+			strcpy((char*)allocPtr, stringParts[0]);
+
+			//Grab integer size from stringparts and store
+			localLoaderSec->totalSize = std::atoi(stringParts[1]);
+			localLoaderSec->currentSize = 0;
+		}
+		//Move up one slot
+		localLoaderSec = localLoaderSec + 1;
+		//if we are at the end of the struct and into the game data done loading files
+		if (&loaderGlobs.sectionList[LOADER_MAXSECTIONS] == localLoaderSec) {
+			if (f != nullptr) {
+				Gods98::File_Close(f);
+			}
+			Gods98::File_SetLoadCallback(Loader_FileLoadCallback, (void*)0x0);
+			return;
+		}
+	} while (true);
+}
 
 // <LegoRR.exe @0044e000>
-//void __cdecl LegoRR::Loader_Shutdown(const char* filename);
+void __cdecl LegoRR::Loader_Shutdown(const char* filename)
+{
+	Gods98::File* f;
+	LoaderSection* localLoaderSec;
+
+	//Check if file is null to setup nullptr
+	if (filename == nullptr) {
+		f = nullptr;
+	}
+	else {
+		f = Gods98::File_Open(filename, "w");
+	}
+
+	//Setup localStruct to clear out all memory
+	localLoaderSec = loaderGlobs.sectionList;
+	do {
+		//If the section isnt null print the section to the file and clear the memory
+		if (localLoaderSec->name != nullptr) {
+			if (localLoaderSec->currentSize == 0) {
+				localLoaderSec->currentSize = localLoaderSec->totalSize;
+			}
+			if (f != nullptr) {
+				Gods98::File_PrintF(f, "%s%s%i\n", localLoaderSec->name, "=", localLoaderSec->currentSize);
+			}
+			Gods98::Mem_Free((void*)localLoaderSec->name);
+		}
+		//Bump up iterator unitl done
+		localLoaderSec = localLoaderSec + 1;
+	} while (localLoaderSec < (LoaderSection*)&loaderGlobs.current);
+
+	//if the file isnt null we're done
+	if (f != nullptr) {
+		Gods98::File_Close(f);
+	}
+	//Remove and close load/shutdown screens
+	Gods98::Image_Remove(loaderGlobs.LoadScreen);
+	if (loaderGlobs.ShutdownScreen != nullptr) {
+		Gods98::Image_Remove(loaderGlobs.ShutdownScreen);
+	}
+	return;
+}
 
 // <LegoRR.exe @0044e0a0>
-//void __cdecl LegoRR::Loader_display_loading_bar(const char* section);
+void __cdecl LegoRR::Loader_display_loading_bar(const char* section)
+{
+	LoaderSection* localLoaderSec;
+	int secCntr;
+	//If section is null clear all flags and bails
+	if (section == nullptr) {
+		loaderGlobs.flags = loaderGlobs.flags & (~LOADER_GLOB_FLAG_ENABLED);
```

Comment:
Prefer using compound assignment: `loaderGlobs.flags &= ~LOADER_GLOB_FLAG_ENABLED;`

Same goes for the bitwise OR below: `loaderGlobs.flags |= LOADER_GLOB_FLAG_ENABLED;`